example-document/03-troubleshooting.md

```diff
@@ -0,0 +1,83 @@
+# Troubleshooting {#troubleshooting}
+
+Many pipeline failures are caused by justa few issues. This section describes the debugging process of a failed pipeline.
+
+First of all, to see the pipeline statu:
+1. Go to your repository
+2. Open Actions tab
+3. Open detail of your pipeline run
+4. Find out what job is failing
+5. See detailed logs of failed job
+6. Seach for commonly known error messages using native GitHub log search on top right corner
+
+## Validate YAML documents
+
+YAML formatting is strict, and all errors in this job are related to it. Search for `yaml.scanner.ScannerError` to see what went wrong. Example of the log:
+
+```
+yaml.scanner.ScannerError: mapping values are not allowed in this context
+  in "/__w/istqb-ctal-ta/istqb-ctal-ta/sample-exam-answers/questions.yml", line 562, column 19
+```
+To resolve these issues:
+1. Navigate to affected file (`sample-exam-answers/questions.yml` line 562, position 19 in this case)
+2. Resolve the formatting issue which could be cause by several things. The most common are:
+  * wrong indents - fix by adding 2 spaces per indent level
+  * using colon `:` or hashtag `#` in the text - fix by adding the text in quotes `"` or `'`
+  * missing pipe `|` in right after the key name to allow multiline inputs - fix by adding pipe `|` before multiline markdown input
+
+## Produce PDF documents
+
+### Undefined references and citations
+
+The most common failure is one related to references. Search for `Undefined refs and citations` to see what went wrong. Example of the log:
+
+```
+Latexmk: ====Undefined refs and citations with line #s in .tex file:
+  Reference `section:analyze-sut-to-determine-the-appropriate-tas' on page 21 undefined on input line 37
+  Reference `section:apply-layering-of-taf' on page 22 undefined on input line 38
+  Reference `section:apply-design-principles-and-design-patterns-in-ta' on page 24 undefined on input line 38
+  Reference `section:apply-different-approaches-for-automating-tcs' on page 47 undefined on input line 43
+  Reference `section:apply-layering-of-taf' on page 48 undefined on input line 43
+  Reference `section:explain-which-factors-support-and-affect-tas-maintainability' on page 50 undefined on input line 43
+```
```

Comment:
This code example overflows:

![image](https://github.com/user-attachments/assets/b54dac2b-4ee2-449c-8604-88c090af89b4)

In general, code examples should not exceed 90 characters per line.

Here, I suggest line-breaking the logs:

````suggestion
```
Latexmk: ====Undefined refs and citations with line #s in .tex file:
  Reference `section:analyze-sut-to-determine-the-appropriate-tas'
    on page 21 undefined on input line 37
  Reference `section:apply-layering-of-taf'
    on page 22 undefined on input line 38
  Reference `section:apply-design-principles-and-design-patterns-in-ta'
    on page 24 undefined on input line 38
  Reference `section:apply-different-approaches-for-automating-tcs'
    on page 47 undefined on input line 43
  Reference `section:apply-layering-of-taf'
    on page 48 undefined on input line 43
  Reference `section:explain-which-factors-support-and-affect-tas-maintainability'
    on page 50 undefined on input line 43
```
````

Reply:
Fixed